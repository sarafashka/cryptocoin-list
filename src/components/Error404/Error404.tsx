import React from 'react';
import Error404Svg from './Error404svg';
// import './Error404.css';

const Error404: React.FC = () => {
  return (
    <>
      <div className="error404" role="error404">
        <Error404Svg />
        <div className="message__box">
          <h1>404</h1>
          <p>Page not found</p>
        </div>
      </div>
    </>
  );
};

export default Error404;
