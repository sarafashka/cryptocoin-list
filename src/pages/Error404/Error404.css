
.error404 {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  gap: 50px;
}

.message-box {
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
  justify-content: center;
  gap: 40px;
  min-width: 40vh;
  height: 100%;;
}

.message-box h1 {
  font-size: 100px;
  line-height: 46px;
  color: #1876d2;

}

.message-box p {
  font-size: 20px;

}


#Polygon-1 , #Polygon-2 , #Polygon-3 , #Polygon-4 , #Polygon-4, #Polygon-5 {
  animation: float 1s infinite ease-in-out alternate;
}
#Polygon-2 {
  animation-delay: .2s;
}
#Polygon-3 {
  animation-delay: .4s;
}
#Polygon-4 {
  animation-delay: .6s;
}
#Polygon-5 {
  animation-delay: .8s;
}

@keyframes float {
  100% {
    transform: translateY(20px);
  }
}

.error-404-svg {
  width: 40vh;
  height: auto;
}

@media (max-width: 767px) {
  .error-404-svg {
    position: absolute;
    top: 50%;
    left: 50%;
    z-index: 1;
    transform: translate(-50%, -50%);
  }
  .message-box {
    position: relative;
    z-index: 2;
    background-color: rgba(249, 248, 248, 0.7);
  }
}

