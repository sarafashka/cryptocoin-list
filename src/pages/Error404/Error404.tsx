import React from 'react';
import Error404Svg from './Error404svg';
import AppRoutes from '../../constants/routes';
import './Error404.css';
import { useNavigate } from 'react-router-dom';

const Error404: React.FC = () => {
  const navigate = useNavigate();
  return (
    <>
      <div className="error404" role="error404">
        <Error404Svg />
        <div className="message__box">
          <h1>404</h1>
          <p>Page not found</p>
          <button
            role="button"
            className="message__button"
            onClick={() => navigate(AppRoutes.HOME)}
          >
            Go to the main page
          </button>
        </div>
      </div>
    </>
  );
};

export default Error404;
