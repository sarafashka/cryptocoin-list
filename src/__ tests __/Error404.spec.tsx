import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Error404 from '../pages/Error404/Error404';

describe('Error404 component', () => {
  it('renders correctly', () => {
    render(
      <MemoryRouter>
        <Error404 />
      </MemoryRouter>
    );
    expect(screen.getByRole('error404')).toMatchSnapshot();
  });
});
