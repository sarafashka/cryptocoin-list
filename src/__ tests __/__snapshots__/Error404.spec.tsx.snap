// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 component renders correctly 1`] = `
<div
  class="error404"
  role="error404"
>
  <svg
    height="500px"
    version="1.1"
    viewBox="0 0 837 1045"
    width="380px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
      id="Page-1"
      stroke="none"
      stroke-width="1"
    >
      <path
        class="polygon1"
        d="M353,9 L626.664028,170 L626.664028,487 L353,642 L79.3359724,487 L79.3359724,170 L353,9 Z"
        id="Polygon-1"
        stroke="#b982ff"
        stroke-width="6"
      />
      <path
        d="M78.5,529 L147,569.186414 L147,648.311216 L78.5,687 L10,648.311216 L10,569.186414 L78.5,529 Z"
        id="Polygon-2"
        stroke="#b982ff"
        stroke-width="6"
      />
      <path
        d="M773,186 L827,217.538705 L827,279.636651 L773,310 L719,279.636651 L719,217.538705 L773,186 Z"
        id="Polygon-3"
        stroke="#b982ff"
        stroke-width="6"
      />
      <path
        d="M639,529 L773,607.846761 L773,763.091627 L639,839 L505,763.091627 L505,607.846761 L639,529 Z"
        id="Polygon-4"
        stroke="#b982ff"
        stroke-width="6"
      />
      <path
        d="M281,801 L383,861.025276 L383,979.21169 L281,1037 L179,979.21169 L179,861.025276 L281,801 Z"
        id="Polygon-5"
        stroke="#b982ff"
        stroke-width="6"
      />
    </g>
  </svg>
  <div
    class="message__box"
  >
    <h1>
      404
    </h1>
    <p>
      Page not found
    </p>
    <button
      class="message__button"
      role="button"
    >
      Go to the main page
    </button>
  </div>
</div>
`;
